### ISSUE TITLE
When and how to update?

### ISSUE THREAD

**Issue by danwie** [Georift/install-jellyfin-tizen#25]
I have already installed using this project a couple of months ago. I updated to jellyfin server 10.9.1. I'm now curious of when and how I can update to latest version of the client. 

1. How can I find out what version of jellyfin tizen I have installed.

2. Where can I see the current version of jellyfin tizen?

3. Does the install using docker method command works when updating also? Or do I need to remove the app from my TV first?

Sorry if these questions have already been answered.

**Comment by akrasic** [comment]
I can answer some of the questions:

For the docker command to work, at least for me, was adding `--ulimit nofile=1024:65536` to the command, so the full command would be:
`docker run --rm --ulimit nofile=1024:65536  georift/install-jellyfin-tizen $TV_IP_ADDRESS` 

Because this error:
`library initialization failed - unable to allocate file descriptor table - out of memory/tizen-studio/tools/ide/bin/tizen.sh: line 143:  `

You need to remove Jellyfin client from the TV before, I had it installed, updated has errored out on :
`app_id[AprZAARz4r.Jellyfin] install failed[118, -11], reason: Author certificate not match :`


After that everything went smoothly, app (re)installed, and working fine. 
Hope it helps

**Comment by leekslk** [comment]
@akrasic thanks for the helpful tip. was the jellyfin-tizen client updated (to 10.9.1 web related) or just some of the dependencies?

**Comment by akrasic** [comment]
Yes, it was updated to 10.9.1, it pulls  it from here https://github.com/jeppevinkel/jellyfin-tizen-builds . You basically get the web client. 
IMO much better than the Android TV client I have on Shield.

**Comment by jeppevinkel** [comment]
> Yes, it was updated to 10.9.1, it pulls it from here https://github.com/jeppevinkel/jellyfin-tizen-builds . You basically get the web client. IMO much better than the Android TV client I have on Shield.

I'm curious. Do you have it working with 10.9.1 on your TV? Me as well as others can't get it running currently. Some people seem to get a header issue right when launching, and mine freezes after I sign in.

I don't currently have a definite source of the issue, so while I technically have 10.9.1 and now 10.9.2 builds in my repository, I don't currently recommend using them.

**Comment by akrasic** [comment]
I'm using 10.9.1 still, haven't upgraded yet. I'll see when I have time in the day to upgrade the server and client to 10.9.2 and report back.

**Comment by akrasic** [comment]
@jeppevinkel  Sorry it took bit longer than I expected, I've upgraded the server and TV app to 10.9.2. 
Login works and playback works. 
Worth noting is - I'm connecting to a machine on local network, and using the LAN IP + port to connect directly to Jellyfin, without any reverse proxies between.

**Comment by jeppevinkel** [comment]
> @jeppevinkel Sorry it took bit longer than I expected, I've upgraded the server and TV app to 10.9.2. Login works and playback works. Worth noting is - I'm connecting to a machine on local network, and using the LAN IP + port to connect directly to Jellyfin, without any reverse proxies between.

I had a similar setup when I faced the issues. I have since gotten it to work thanks to a fix being pushed to the release-10-9-z branch of the jellyfin-web repository. It appears that some TVs were just unaffected by the issue. Probably due to different version of firmware or software on the TV.

**Comment by jeppevinkel** [comment]
Selecting The `Jellyfin-prerelease` version with this installer should now work for everyone since the latest release of that includes the fix.

**Comment by jeppevinkel** [comment]
> I can answer some of the questions:
> 
> For the docker command to work, at least for me, was adding `--ulimit nofile=1024:65536` to the command, so the full command would be: `docker run --rm --ulimit nofile=1024:65536 georift/install-jellyfin-tizen $TV_IP_ADDRESS`
> 
> Because this error: `library initialization failed - unable to allocate file descriptor table - out of memory/tizen-studio/tools/ide/bin/tizen.sh: line 143: `
> 
> You need to remove Jellyfin client from the TV before, I had it installed, updated has errored out on : `app_id[AprZAARz4r.Jellyfin] install failed[118, -11], reason: Author certificate not match :`
> 
> After that everything went smoothly, app (re)installed, and working fine. Hope it helps

I should also add. For any releases after https://github.com/jeppevinkel/jellyfin-tizen-builds/releases/tag/2024-05-20-1647 and going forward, you will no longer need to remove the old Jellyfin client when updating. I got the certificate stuff sorted out so it now uses a consistent certificate between builds which allows you to seamlessly update and downgrade as you see fit with any releases from that point on.

You will need to uninstall any releases older than that to get it though.

**Comment by danwie** [comment]
Is there a way to update to my TV automatically?

**Comment by leekslk** [comment]
not sure what you meant by automatically. Whenever jellyfin releases an update or when you make server side updates?

**Comment by danwie** [comment]
> not sure what you meant by automatically. Whenever jellyfin releases an update or when you make server side updates?

Whenever there is a new jellyfin-tizen-build, or maby once a week. It seems my TV stays in developer mode. Maby there is already a script that will do the docker run once a week, with some configuration options?

**Comment by Georift** [comment]
This project doesn't attempt automatic updates, you could certainly use it with a cron job to achieve something similar if you can guess when the TV will be on and time the cron job.

I'll leave this issue open as a reminder to summarize these topics and put them into a FAQ.

**Comment by Novsinclair** [comment]
Worked for 2024 model QE83S90DAEXCE

1. Unpack jellyfin.wgt with WinRAR to folder
2. Create new certificate with DUID of your TV (Can be found in device manager)
3. re-sign package with tizen package -t wgt -s YOUR_NAME_CERT_PROFILE -- C:\SAMS\tizen-studio\tools\ide\bin\YOUR_FOLDER
4.  Install it with tizen install -n YOUR_NEW.wgt -t QE83S90DAEXCE
